internal/main.go

```diff
@@ -233,47 +145,138 @@ func RunTests(opts RunTestsOpts) error {
+	if testErr != nil {
+		return ErrTestRunIgnore
+	}
+
+	if covErr != nil {
+		return covErr
 	}
 
 	// Open html coverage report
 	if opts.FlagCoverReport {
 		shCmd("go", shArgs{"tool", "cover", "-html=" + coverProfile}, "")
 	}
+
 	return nil
 }
 
-func printCoverage(path string, lineOut func(...string)) (float64, error) {
-	var coverage float64
+// Add dummy test files to packages with no tests
+func initEmpty(testPath string, excludes []string) (newFiles []string, packages []Package, err error) {
+	allPkgs := []string{}
+	allPkgsMap := map[string]Package{}
+
+	{
+		pkgChan := make(chan Package)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			for p := range pkgChan {
+				allPkgs = append(allPkgs, p.ImportPath)
+				allPkgsMap[p.ImportPath] = p
+			}
+			wg.Done()
+		}()
+		err := shJSONPipe("go", shArgs{"list", "-json", testPath}, "", pkgChan, io.Discard)
+		wg.Wait()
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	skippedPkgs := []string{}
+
+	{
+		testPkgs, _, err := excludePackages(allPkgs, excludes)
+
+		if err != nil {
+			return nil, nil, err
+		}
+
+		goListOutput := make(chan TestEvent)
+
+		var wg sync.WaitGroup
+		wg.Add(1)
+
+		go func() {
+			for o := range goListOutput {
+				if o.Action == "skip" && o.Test == "" {
+					skippedPkgs = append(skippedPkgs, o.Package)
+				}
+			}
 
-	// now read proper code coverage
+			wg.Done()
+		}()
+
+		testArgs := shArgs{"test"}
+		testArgs = append(testArgs, "-list", ".")
+		testArgs = append(testArgs, "-json")
+		testArgs = append(testArgs, testPkgs...)
+		err = shJSONPipe("go", testArgs, "", goListOutput, io.Discard)
+		wg.Wait()
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
+	defer func() {
+		if err != nil {
+			for _, f := range newFiles {
+				os.Remove(f)
+			}
+		}
+	}()
+
+	for _, importPath := range skippedPkgs {
+		p, ok := allPkgsMap[importPath]
+		if !ok {
+			return nil, nil, fmt.Errorf("package %q not found in map", importPath)
+		}
+
+		// this is all that's need to be printed to be a valid test
+		textToWrite := "package " + p.Name + "\n"
+
+		file, err := os.CreateTemp(p.Dir, "dummy_*_test.go")
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// we know we can write to dummy_test, because there is no test in the package
+		err = os.WriteFile(file.Name(), []byte(textToWrite), 0o666)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		newFiles = append(newFiles, file.Name())
+		packages = append(packages, p)
+	}
+	return newFiles, packages, nil
+}
+
+func printCoverage(path string, lineOut func(...string)) (float64, error) {
+	regexTotalCov := regexp.MustCompile(`^total:\s+\(statements\)\s+(\d{1,3}\.\d{1,2}%)`)
 	goCoverOutput := make(chan string)
 
 	var wg sync.WaitGroup
 	wg.Add(1)
 
+	var coverage float64
 	go func() {
 		for line := range goCoverOutput {
 			if strings.HasPrefix(line, "total:") {
-				// a bit of a hack but it works
-				line = strings.Trim(line, "total: (statements) %\t\n")
```

Comment:
linter complained about this. cutset should not have duplicate characters.

Reply:
Ah but it still works, sorry